### ISSUE TITLE
Missing parameter in ProductsService.GetProductOrderBookAsync(ProductType productPair).

### ISSUE THREAD

**Issue by JensKDK** [dougdellolio/gdax-csharp#50]
Hi, and thank you for all the work.

According to the API documentation, the "Get Product Order Book" service, allows you to specify a "Level". If you wish to maintain a complete Orderbook (using the WebFeed), which I do, I need to start by calling GetProductOrderBookAsync with "Level=3", which returns the entire orderbook non aggregated. Afterwards you keep it updated from the webfeed. 

In ProductsService.GetProductOrderBookAsync no level parameter is sent, and I assume it then defaults level to 1, which only returns the best bid and ask.

Would you please add this parameter to this function, so we can all enjoy full functionality.
Thank you very much.

Regards,
Jens.

**Comment by dougdellolio** [comment]
@JensKDK 

I just opened this PR here: https://github.com/dougdellolio/gdax-csharp/pull/51

Let me know if you are good with this change and I will merge and release with next version.

**Please note there is a breaking change as the response `ProductsOrderBookResponse` is now returning a `IEnumerable<IEnumerable<string>>` as it was returning `IEnumerable<IEnumerable<decimal>>` before. You will need to parse as decimal for levels 1 and 2.

thanks!

**Comment by JensKDK** [comment]
@dougdellolio

Thank you for your VERY quick response. I couldn't be happier about the changes. Look forward to the release! Thank you!  

Regards,
Jens.

**Comment by dougdellolio** [comment]
@JensKDK released in v1.0.8. thx 😃

**Comment by JensKDK** [comment]
Thank you!! :-)

**Comment by JensKDK** [comment]
@dougdellolio 

Hi again,

After testing, I have to re-open this. Sorry.

Looking at the service response: https://api.gdax.com/products/BTC-EUR/book?level=3 ...

In ProductsService. ConvertProductOrderBookResponse:

1) 
It does not properly convert the strings to decimal. The strings contains a ".", which should be a "," for proper conversion. "." are disregarded, and "10.00" is converted to "1000". 
Using "Convert.ToDecimal(askArray[2].Replace(".", ","))", or similar, solves it. 
"Price" and "Size" needs this conversion as well.

2)
As you can see in the service result "NumberOfOrders" does not exist. I don't understand why you put it in there? Array[0] is price, 1 is Size and 2 is OrderID.

Thank you.

Regards, Jens,

**Comment by dougdellolio** [comment]
@JensKDK sorry to hear its not working. I think this is a CultureInfo problem?

what happens if you were to try this?

`Convert.ToDecimal(askArray[2], CultureInfo.InvariantCulture)`

also, when requesting level 3 the `OrderId` should be set on the Bid/Ask object but the `NumberOfOrders` will be null. Please let me know if that is not the case

**Comment by JensKDK** [comment]
@dougdellolio 

Sorry, while I've been programming professionally for many years, I'm pretty new to C# and .NET.

When using CultureInfo.InvariantCulture it works :-)
NumberOfOrders become null because it cannot convert bidArray[2] to a decimal. Is there a reason to even try the conversion. It will never work. Also, it's set to null default in Quote.

Thank you for your very quick replies. Look forward to the release of the fix.

Regards, Jens.

**Comment by dougdellolio** [comment]
@JensKDK no worries, and glad to hear setting the CultureInfo works! I will make this change. 

The GDAX API returns the following for level 1 and level 2:

```
"bids": [
        [ price, size, num-orders ],
        [ "295.96", "4.39088265", 2 ],
        ...
    ],
```

and this for level 3:

````
    "bids": [
        [ price, size, order_id ],
        [ "295.96","0.05088265","3b0f1225-7f84-490b-a29f-0faef9de823a" ],
        ...
    ],
````

The only difference is the last element of the collection (bidArray[2]). The logic currently will set the `NumberOfOrders`  property to a value and the `OrderId` to null if the book level specified is 1 or 2. If the book level is 3, `NumberOfOrders` will be null and the `OrderId` will be set. Therefore, the conversion always happens but the actual property the value is set to changes based on the book level. This logic is within the `ConvertProductOrderBookResponse` method in `ProductsService`.

Does this make sense? Also if you feel there is a better way, I am open to suggestions

**Comment by JensKDK** [comment]
@dougdellolio 

Ahh, level 1 and 2 of course. Yes, that is fine. No reason to make separate responses.

Thank you again :-)

**Comment by dougdellolio** [comment]
please check this out and verify it works: 

https://github.com/dougdellolio/gdax-csharp/pull/52

if you're 👍 ill merge and release. thanks for your help!

**Comment by JensKDK** [comment]
@dougdellolio 

I joined GitHub yesterday, and honestly have no idea how to pull it down and test it. I can check the code however, and since the only problem was the CultureInfo, and you've fixed that, it must work. 

Thank you very much!

**Comment by dougdellolio** [comment]
@JensKDK released with v1.0.9. Should appear in NuGet shortly. Feel free to open up another issue if its still not working (fingers crossed!)

**Comment by JensKDK** [comment]
@dougdellolio 

Tested! I works perfectly! :-)

Thank you very much!
